### DISCUSSION TITLE
MQTT & NTP Data format

### DISCUSSION THREAD

**Discussion by Kalored** [frickelzeugs/FingerprintDoorbell#25] [Q&A]
Hey, 

This is an awesome project which came up exactly the right time when I wanted to replace my old Conrad Fingerprint Sensor which is now +10years old and not working well anymore.

I got all the parts in the mail last weekend and put it together. All is working fine so far, I can scan fingers, enroll fingers etc. 
Whats not really clear to me which format the MQTT data have to be put into the box?
My iobroker mqtt instance is running on 192.168.xxx.4 with the Port 1893 (not standard Port) . When I try to enter the data in the format "192.168.xxx.4:1893" it doesnt work. Also when I add "mqtt://" or "tcp://" before the whole address it doesnt work. I have bunch of other devices running on the same instance (e.g. Tasmota devices), so I know for sure this instance is up and working fine. Is there any trick to get the port and address in there? 

Other questions would be how is the pattern with the NTP server? I can change it to a different Server, but it always takes the UTC time and not GMT+1 for e.g. Germany. 

As a feature request it would be nice to deactivate/change the blue breathing light in standard. I dont want to attract more attention than necessary on that button at night.

**Comment by frickelzeugs** [comment]
Unfortunately the port 1883 is hard coded at the moment, so only way to change this is to edit the line 666 (that was not intentional 😆) of main.cpp and make your own build. But I take this as a feature request to parse the port number out of the string.

A NTP-Server always delivers UTC time. Conversion to local time is the clients job. So what you are looking for is timezone support in FingerprintDoorbell and this isn't implemented, only UTC supported yet.

**Reply by Kalored** [reply]
Thanks for your quick answer. I will try to edit line 666 then and compile my own build. I hope to get it work. :)

**Reply by nobbi1991** [reply]
I created a pull request to change the MQTT port via the web GUI.

**Comment by TheMaskedDeveloper** [comment]
Regarding your feature request to deactivate/change the blue breathing light in standard mode:
I implemented it and created a pull request as I just wanted the same feature. The color and sequence can be changed in the settings menu. Also the other colors for the newer R503 sensor can be selected there. For me it's white / on (non breathing) now.
(And Line 666 moved to 693 :-) )

**Reply by Kalored** [reply]
Awesome! Thanks for the info. When is a new release planned where this is included? :)

**Reply by TheMaskedDeveloper** [reply]
At the moment it's just a pull request. I think frickelzeugs will have a look at it and give feedback for changes, etc. Then he decides if the feature is merged into the master at some point in time.
But you can try out the feature by checking out the PR (https://github.com/frickelzeugs/FingerprintDoorbell/pull/26) or the fork (https://github.com/TheMaskedDeveloper/FingerprintDoorbell.git)
